hpack.go

```diff
@@ -392,6 +395,9 @@ func appendInt(dst []byte, bits uint8, index uint64) []byte {
 func readString(dst, b []byte) ([]byte, []byte, error) {
 	var n uint64
 	var err error
+	if len(b) < 1 {
+		return b, dst, errors.New("invalid data")
```

Comment:
Can you return `b, dst, errors.New("no bytes left reading a string. Malformed data?")`? Maybe we should standardize the error, but lets leave it like that